### ISSUE TITLE
Don't add duplicated values for the same `Content-Type` header

### ISSUE THREAD

**Issue by massdonati** [bustoutsolutions/siesta#246]
Hello,
I'm currently seeing that if we configure the service as follows
```
service().configure { config in
    config.headers["Content-Type"] = "application/json"
}
```
when creating a request with the `request(_:json:)` api the `Content-Type` header key is set to `application/json,application/json`.

**Comment by pcantrell** [comment]
Sorry for the lack of response; I've been focusing on my limited Siesta time on getting the cache improvements out the door.

Out of curiosity, does changing `underlyingRequest.addValue` to `underlyingRequest.setValue` fix the issue for you?

**Comment by massdonati** [comment]
Hello, no worries and thanks for the response! That was my first implementation and yes it did fix the issue, but by checking if the contentType is contained in the `underlyingRequest` header field I'm being more conservative: for example if the request has different and multiple content types. Instead if I were to use `setValue` i would replace the entire value.

**Comment by pcantrell** [comment]
Hmm, you are right to ask these questions. I don't think _multiple_ content-type headers are ever correct — but if a `request(_:method:contentType:)` argument and configuration _both_ specify a content type, which should win? I tend to think the argument should, since it is more local, more specific to the individual request.

**Comment by massdonati** [comment]
I see so you suggest to basically replace whatever content-type the configuration set with the method's `contentType` parameter, correct? If so I'll make the changes.
As a side note, unrelated with this issue, I'm wondering if the `contentType` parameter in 
```
public func request(
            _ method:        RequestMethod,
            text:            String,
            contentType:     String = "text/plain",
            encoding:        String.Encoding = String.Encoding.utf8,
            requestMutation: @escaping RequestMutation = { _ in })
        -> Request
```
and 
```
public func request(
            _ method:        RequestMethod,
            json:            JSONConvertible,
            contentType:     String = "application/json",
            requestMutation: @escaping RequestMutation = { _ in })
        -> Request
```
can go away and be hard coded? 
Anyway thanks for the response.

**Comment by massdonati** [comment]
@pcantrell I just update the pr.

**Comment by pcantrell** [comment]
Thanks, Massimo! I'm just getting to merging this now; thanks for your patience. I spent some time scratching my chin, and decided that yes, the discussion above is correct:

- The request methods that set `Content-Type` should override the header value, not add it.
- The override order (first one wins) is:
  - configured request mutation
  - ad hoc request mutation
  - default content type for request helper method
  - header set in configuration

I [added some tests to the PR](https://github.com/bustoutsolutions/siesta/pull/247/commits/b5419151670423a939ecf75df8f2e1c2feb9808f) to reflect this.

---

> As a side note, unrelated with this issue, I'm wondering if the contentType parameter in [various request helpers] can go away and be hard coded? The reasoning for that is that I think is more error prone letting the user the freedom to specify an application/xml even if for example is sending json.

No, because many APIs require specialized content types for familiar data type. In fact, the original application that Siesta came from works with an API that wanted [`Content-Type: application/hal+json `](http://stateless.co/hal_specification.html).

**Comment by pcantrell** [comment]
Fixed in #247.
